src/Executor/CommandLine.cs

```diff
@@ -0,0 +1,1857 @@
+        {
+            var help = false;
+            CommandLineParser parser = new CommandLineParser(args, setupContent);
+            parser.DefineOptionalQualifier("?", ref help, "Print this help guide.", null, null);
+
+            try
+            {
+                // RawPrint the help.
+                if (parser.HelpRequested != null)
+                {
+                    parser._skipParameterSets = true;
+                    parser._skipDefinitions = true;
+                }
+                parseBody(parser);
+                if (parser.HelpRequested != null)
+                {
+                    helpHandler(parser);
+                    return false;
+                }
+                parser.CompleteValidation();
+                return true;
+            }
+            catch (CommandLineParserException e)
+            {
+                errorHandler(parser, e);
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Qualifiers are command line parameters of the form -NAME:VALUE where NAME is an alphanumeric name and
+        /// VALUE is a string. The parser also accepts -NAME: VALUE and -NAME VALUE but not -NAME : VALUE For
+        /// boolan parameters, the VALUE can be dropped (which means true), and a empty string VALUE means false.
+        /// Thus -NAME means the same as -NAME:true and -NAME: means the same as -NAME:false (and boolean
+        /// qualifiers DONT allow -NAME true or -NAME false).
+        ///
+        /// The types that are supported are any type that has a static 'Parse' function that takes a string
+        /// (this includes all primitive types as well as DateTime, and Enumerations, as well as arrays of
+        /// parsable types (values are comma separated without space).
+        ///
+        /// Qualifiers that are defined BEFORE any parameter sets apply to ALL parameter sets.  qualifiers that
+        /// are defined AFTER a parameter set will apply only the the parameter set that preceeds them.
+        ///
+        /// See code:#DefiningParametersAndQualifiers
+        /// See code:#Overview
+        /// <param name="name">The name of the qualifier.</param>
+        /// <param name="retVal">The place to put the parsed value</param>
+        /// <param name="helpText">Text to print for this qualifier.  It will be word-wrapped.  Newlines indicate
+        /// new paragraphs.</param>
+        /// </summary>
+        public void DefineOptionalQualifier<T>(string name, ref T retVal, string helpText, string defaultValue, List<string> legalValues)
+        {
+            object obj = DefineQualifier(name, typeof(T), retVal, helpText, false, defaultValue, legalValues);
+            if (obj != null)
+                retVal = (T)obj;
+        }
+        /// <summary>
+        /// Like code:DeclareOptionalQualifier except it is an error if this parameter is not on the command line.
```

Comment:
For now I'm just modifying the methods that we care about on this class.
Maybe later we can do a clean up